src/_zkapauthorizer/tests/test_storage_protocol.py

```diff
@@ -192,11 +206,20 @@ class ShareTests(TestCase):
     def setUp(self):
```

Comment:
> That said, it feels like testtools + fixtures are fighting hypothesis enough that it would make more sense to try and find a better combined solution

Yes, agreed.  Until now I've had the vague idea that it is possible to integrate them better _somehow_ but I hadn't really investigated how.  From the docs you linked, I see the hooks Hypothesis offers now and maybe those provide a path (eg `inner_test`?).  But that's probably upstream testtools development that needs to happen... and I don't know how likely the testtools project is to accept any contributions at this point.  I'm also not super motivated to work on this these days.  Maybe at some future point when some other priorities have eased up a bit...

So anyway I'm leaving this part of the PR as-is for now